rfcs/0009-code-review.md

```diff
@@ -0,0 +1,173 @@
+---
+created-on: 2021-04-27
+freshness: 2021-04-27
+owner: @dj2
+status: proposal
+tags:
+  - software development
+  - practices
+---
+
+[La version française suit.](#résumé)
+
+# Code Review
+
+## Summary
+Code review is an integral part of how we work at CDS. Reviewing code
+provides an opportunity for other developers to understand the code. It
+provides a second set of experiences to shape the code. This RFC lays
+out considerations for reviewing code at CDS.
+
+## Motivation
+A shared set of considerations helps ease both the burden of being
+reviewed and that of being the reviewer. Code review is inherently
+subjective, different reviewers focus on different things. This can make
+the review system seem opaque. Shared considerations help to ease that
+subjectivity by giving guide rails to make reviews more consistent.
+
+## Proposal
+Below is a list of considerations these are divided into principles and
+technical consideration. Focus on the principles and then lead into the
+technical considerations.
+
+### General guidelines
+ - Code reviews are required at CDS.
+   - Each review should have a technical reviewer.
+   - Add Design, Research, Product Management, Service Owner, as needed.
+ - Tend towards questions rather then statements.
+   - The reviewer may not understand all the considerations a reviewee
+     put into the code.
+   - The reviewee may not understand the reviewers past experience with
+     the code.
+ - [Be nice](https://qz.com/work/625870/after-years-of-intensive-analysis-google-discovers-the-key-to-good-teamwork-is-being-nice/)
+   - Read the CDS [Code of Conduct](https://github.com/cds-snc/.github/blob/master/CODE_OF_CONDUCT.md)
+ - Escalate if the review is blocked.
+   - The goal of the review is that the code looks good to the reviewer.
+     It doesn't have to be perfect, it doesn't have to be how the
+     reviwere would write it. Just, good.
+ - In general authors should merge their own PRs.
```

Comment:
I would also highlight that if someone else but the reviewee presses the 'squash and merge' on the PR the reviewee loses all credit to their contribution and the merged commit is in the name of the reviewer (or person who pressed 'squash and merge'.

Reply:
Good idea, added. Better?

Reply:
Thanks :)